KsApi/combine/CombineTestObserver.swift

```diff
@@ -27,4 +33,161 @@ public final class CombineTestObserver<Value, Error: Swift.Error> {
     }
     .store(in: &self.subscriptions)
   }
+
+  /// Get all of the next values emitted by the signal.
+  public var values: [Value] {
+    var values: [Value] = []
+    for event in self.events {
+      switch event {
+      case let .value(v):
+        values.append(v)
+      default: break
+        // do nothing
+      }
+    }
+
+    return values
+  }
+
+  /// Get the last value emitted by the signal.
+  public var lastValue: Value? {
+    return self.values.last
+  }
+
+  /// `true` if at least one `.Next` value has been emitted.
+  public var didEmitValue: Bool {
+    return self.values.count > 0
+  }
+
+  /// The failed error if the signal has failed.
+  public var failedError: Error? {
+    var errors: [Error] = []
+    for event in self.events {
+      switch event {
+      case let .error(e):
+        errors.append(e)
+      default: break
+        // do nothing
+      }
+    }
+
+    assert(
+      errors.count <= 1,
+      "I'm pretty sure a Combine publisher can only ever emit one error. If this fails, we've learned something new today."
+    )
+
+    return errors.last
+  }
+
+  /// `true` if a `.Failed` event has been emitted.
+  public var didFail: Bool {
+    return self.failedError != nil
+  }
+
+  /// `true` if a `.Finished` event has been emitted or a `.Failed` event has been ommitted
+  public var didComplete: Bool {
+    return self.events.contains { event in
+      switch event {
+      case .finished:
+        return true
+      case .error:
+        return true
```

Comment:
This is one difference. A _failed_  publisher in Combine is 'dead' and complete, whereas I believe that's not true in ReactiveSwift.